Realm/Tests/MigrationTests.mm

```diff
@@ -1398,6 +1398,390 @@ - (void)testModifyPrimaryKeyInMigration {
     }
 }
 
+- (void)testChangeEmptyTableFromTopLevelToEmbedded {
+    RLMProperty *intProperty = [[RLMProperty alloc] initWithName:@"intProperty"
```

Comment:
Rather than manually defining the base schema, this should just define a normal subclass and use that for the initial schema (or reuse an existing type if possible). The general pattern these tests should take is obtain schema from classes with `schemaForObjectClass:`, mutate a second copy of the schema to match the thing being tested, then call `assertMigrationRequiredForChangeFrom:to:` which initializes the Realm with the initial schema, verifies that the schema version has to be bumped to migrate to the new schema, and then verifies that the tables are in the correct state after migrating to the new schema.

The tests which actually have a migration block should use `createTestRealmWithSchema :` + `migrateTestRealmWithBlock:`